submissions/VBystrov/task_js_oop/app.js

```diff
@@ -0,0 +1,169 @@
+const numEnemies = 5;
+const enemyMinSpeed = 50;
+const enemyMaxSpeed = 250;
+const enemyMinStartX = -700;
+const enemyMaxStartX = 0;
+const enemyMinStartY = 60;
+const enemyMaxStartY = 395;
+
+const playerStartX = 200;
+const playerStartY = 400;
+const playerSpeed = 10;
+
+function randomMinMax(min, max) {
+  return Math.floor(Math.random() * (max - min) + min);
+}
+
+const Enemy = function () {
+  this.sprite = 'images/enemy-bug.png';
+  Resources.load(this.sprite);
+};
+
+Enemy.prototype.update = function (dt) {
+  const resetLine = 640;
+  this.x += this.speed * dt;
+  if (this.x > resetLine) {
+    this.startPosition();
+  }
+};
+
+Enemy.prototype.startPosition = function () {
+  this.x = randomMinMax(enemyMinStartX, enemyMaxStartX);
+  this.y = randomMinMax(enemyMinStartY, enemyMaxStartY);
+  this.speed = randomMinMax(enemyMinSpeed, enemyMaxSpeed);
+};
+
+Enemy.prototype.render = function () {
+  ctx.drawImage(Resources.get(this.sprite), this.x, this.y);
+};
+
+const Player = function () {
+  this.speed = playerSpeed;
+  this.sprite = 'images/char-horn-girl.png';
+  Resources.load(this.sprite);
+};
+
+Player.prototype.update = function (dx = 0, dy = 0) {
+  const minX = -10;
+  const maxX = 420;
+  const minY = -10;
+  const maxY = 400;
+
+  this.x += dx * this.speed;
+  this.y += dy * this.speed;
+  if (this.x < minX) {
+    this.x = minX;
+  }
+  if (this.x > maxX) {
+    this.x = maxX;
+  }
+  if (this.y < minY) {
+    this.y = minY;
+  }
+  if (this.y > maxY) {
+    this.y = maxY;
+  }
+};
+
+Player.prototype.render = function () {
+  ctx.drawImage(Resources.get(this.sprite), this.x, this.y);
+};
+
+Player.prototype.handleInput = function (direction) {
+  const d = 1;
+  let dx = 0;
+  let dy = 0;
+  switch (direction) {
+    case 'left':
+      dx = -d;
+      break;
+    case 'up':
+      dy = -d;
+      break;
+    case 'right':
+      dx = d;
+      break;
+    case 'down':
+      dy = d;
+      break;
+    default:
+      break;
+  }
+  this.update(dx, dy);
+};
+
+Player.prototype.checkCollisions = function (enemies) {
+  // coordinates collision
+  const minX = 18;
+  const maxX = 84;
+  const minY = 85;
+  const maxY = 137;
```

Comment:
Worth grouping with other constants. Think of it as of game settings. Most likely user would want have them all together.

Reply:
Done.